server/src/main/java/org/opensearch/indices/recovery/RecoveryTarget.java

```diff
@@ -109,27 +90,15 @@ public class RecoveryTarget extends AbstractRefCounted implements RecoveryTarget
      * @param sourceNode                        source node of the recovery where we recover from
      * @param listener                          called when recovery is completed/failed
      */
-    public RecoveryTarget(IndexShard indexShard, DiscoveryNode sourceNode, PeerRecoveryTargetService.RecoveryListener listener) {
-        super("recovery_status");
+    public RecoveryTarget(IndexShard indexShard, DiscoveryNode sourceNode, ReplicationListener listener) {
```

Comment:
Since it's extending ReplicationTarget, it's using replication listener that is overridden in calls to RecoveryTarget by recoveryListener that implements ReplicationListener interface